src/lib/slide-toggle/slide-toggle.ts

```diff
@@ -195,10 +206,13 @@ export class MatSlideToggle extends _MatSlideToggleMixinBase implements OnDestro
     // emit its event object to the component's `change` output.
     event.stopPropagation();
 
+    if (!this._dragging) {
+      this.toggleChange.emit();
+    }
```

Comment:
Can you just move `this.toggleChange.emit()` down below the `this._emitChangeEvent()`? That's basically the same and removes the unnecessary check.

Reply:
This `this.toggleChange.emit()` also fires when `this.defaults.disableToggleValue` is true. When the value is true, the native element's checked status is reverted, and no value change is emitted.

Reply:
Makes sense.